internal/command/state_mv.go

```diff
@@ -42,6 +43,43 @@ func (c *StateMvCommand) Run(args []string) int {
 		return cli.RunResultHelp
 	}
 
+	// If backup or backup-out flags are set
+	// and the state flag is not set, make sure
+	// the backend is local
+	backupFlagSetWithoutStateFlag := c.backupPath != "-" && c.statePath == ""
+	backupOutFlagSetWithoutStateFlag := backupPathOut != "-" && c.statePath == ""
+
+	var setLegacyLocalBackendFlags []string
+	if backupFlagSetWithoutStateFlag {
+		setLegacyLocalBackendFlags = append(setLegacyLocalBackendFlags, "-backup")
+	}
+	if backupOutFlagSetWithoutStateFlag {
+		setLegacyLocalBackendFlags = append(setLegacyLocalBackendFlags, "-backup-out")
+	}
+
+	if len(setLegacyLocalBackendFlags) > 0 {
+		currentBackend, diags := c.backendFromConfig(&BackendOpts{})
+		if diags.HasErrors() {
+			c.showDiagnostics(diags)
+			return 1
+		}
+
+		// If currentBackend is nil and diags didn't have errors,
+		// this means we have an implicit local backend
+		_, isLocalBackend := currentBackend.(backend.Local)
+		if currentBackend != nil && !isLocalBackend {
+			diags = diags.Append(
+				tfdiags.Sourceless(
+					tfdiags.Error,
+					fmt.Sprintf("Invalid command-line flags: %s", strings.Join(setLegacyLocalBackendFlags[:], ", ")),
+					"Command-line flags -backup and -backup-out are legacy options that operate on a local state file only. You must specify a local state file with the -state flag or switch to the local backend.",
```

Comment:
We've been taking a "fix it while you're in the area" approach to gradually moving all of our user-facing language to talk about "command line _options_" rather than "flags", because we'd been historically pretty inconsistent about it. Could you tweak this message to use that noun instead? (Then it'll be more consistent with [the documentation](https://www.terraform.io/docs/cli/commands/state/mv.html).

(The internal symbols are less important because we're using a package called `flag` anyway, so it's pretty hard to get away from that terminology in the code itself! :grinning: )

Reply:
Thank you! This is very helpful. It wasn't clear to me which to use so this is perfect. I've updated all of my changes to use `option` instead of `flag` :]